src/gsy_e/models/market/__init__.py

```diff
@@ -59,7 +59,7 @@ def wrapper(self, *args, **kwargs):
     return wrapper
 
 
-class Market:
+class MarketBase:
```

Comment:
Although both `Market` and `MarketBase` are correct terms, in my opinion we should distinguish them (I personally would still be confused between them)
I propose to name the market which we trade in `EnergyMarket` or something else

Reply:
I think `EnergyMarket` would have the same ambiguity of `MarketBase`, in the sense that it's still easy to mix it with the "area market". I agree that the difference between `Market` and `MarketBase` is not clear though.
However, please let's discuss this on slack and move on with this PR. We talked about the `MarketBase` name a few days ago and we decided to move on with it. Changing it now requires time and more testing, and the rebranding for the "area" -> "market" is quite urgent.

Reply:
This is quite fine for me but I thought that rebranding again after merging this PR won't be possible, imo it's either now or never
Anyway, let's not drag this PR on, lgtm for now

Reply:
The `MarketBase` class is an internal class which is not exposed to end-users, so it can be renamed as we will. The "important" class for the rebranding is the new `Market` class, which will stay unchanged.

Reply:
Renaming `Area` --> `Market`. Old `Market` class could be renamed to `MarketUnit`, the term 'unit' might aid in some clarity that each area(now called market) is split into smaller units based on time slot

Reply:
@faizan2590 thanks. Maybe even `MarketSlot` would be quite straightforward (at least in my opinion). However, I would suggest you to please create a poll on slack and keep this for a separate PR.

Reply:
I would like to raise awareness on the fact that the Market class is not a class that can operate on its own, and therefore should not be constructed. Essentially it is the base class in the market hierarchy, and all other market classes (OneSidedMarket, TwoSidedMarket etc.) inherit from this class. Therefore, `Base` in `MarketBase` is not a random term, it really means that this is a base class and implies that should not be constructed. 
I raised this point because all the suggestions (MarketUnit, MarketSlot, EnergyMarket) ignore this aspect. IMO Base is necessary to be part of this class name. I also agree that it is better to take this discussion to slack since it has nothing to do with the ticket itself.